### ISSUE TITLE
Unexpected partial matches on facetted field

### ISSUE THREAD

**Issue by fabien** [typesense/typesense#125]
## Description

From what I understand, facetted fields are used verbatim, and `filter_by` queries on such fields would not return hits on partially matched terms. As an example (which is similar to what I was experiencing in my app, using hierarchical facets, lvl0, lvl1, lvl2), I'm using this dataset:

https://github.com/typesense/typesense-instantsearch-adapter/blob/master/test/support/populateTypesenseIndex.js

## Steps to reproduce

run populateTypesenseIndex.js

`http://localhost:8108/collections/products/documents/search?q=*&filter_by=categories.lvl2:Phone`

## Expected Behavior

I would expect nothing to be returned, because there are no `categories.lvl2` entries that equal just `Phone`, but for example: `Cell Phones > Cell Phone Accessories > Data Cables` - this causes many irrelevant hits, it seems.

## Actual Behavior

I get many results, for any `categories.lvl2` entry that just **contains** the term `Phone`.

In reality, I would filter by full hierarchical terms, like: 

`Cell Phones > Cell Phone Accessories > Data Cables`

In my app, when I filter by `Clothing > Men` I get matches for items that actually have `Clothing > Women` assigned, which defeats the purpose of course.

## Metadata

**Typsense Version**: 0.14

**OS**: Docker

**Comment by kishorenc** [comment]
Thanks for reporting this issue. When a field is marked as a facet, an internal verbatim index is created separately for faceting operations. The field is also indexed as a regular field with tokenization for searches. The filter is not working as expected because it should use the faceted index and not the search index. Will roll out a fix for this issue.

**Comment by fabien** [comment]
Hi @kishorenc, any update on this? Thanks!

**Comment by kishorenc** [comment]
Working on it. Wanted to make the behavior a bit more generic rather than assuming only exact filter on facet fields.

**Comment by kishorenc** [comment]
@fabien 

Here's the nightly Docker image with the fix: `typesense/typesense:961a069`

A `:=` operator has been introduced for exact filtering -- this operator can be used only on facet-enabled string fields.

Example: 

`category:= Shoe` will only match shoes and not a category like `Shoe Rack`. It also supports multiple filter values, so you can do `category:= [Shoe, Sneaker]`.

Let me know if you face any issues!

**Comment by fabien** [comment]
Hi @kishorenc thanks for this. 

Haven’t been able to give it a try yet. However, I was wondering hos this will integrate with the Instantsearch adapter? Any update planned for that, perhaps?

**Comment by jasonbosco** [comment]
@fabien Yup, working on it

**Comment by jasonbosco** [comment]
@fabien Just published a pre-release version of Instantsearch adapter: https://www.npmjs.com/package/typesense-instantsearch-adapter/v/0.3.0-0. Let me know if you run into any issues.

**Comment by jasonbosco** [comment]
@fabien you also want to use this Typesense Server docker image with InstantSearch adapter - has one more tiny update: `typesense/typesense:0.15.0.rc25`

**Comment by fabien** [comment]
Hi, just started to test `rc25` - ran into some issue where `createMany` seems to return a different result format now:

```javascript
return this.client.documents().createMany(documents).then(function(res) {
  // res used to return success (bool) and num_imported (int) - now res is a JSON string, not an object
  return { success: res.success, count: res.num_imported };
});
```

**Comment by fabien** [comment]
Apart from the issue above (running 0.6.2 of typesense-js) everything seems to be working fine! Thanks for these updates!

**Comment by jasonbosco** [comment]
With Typesense v0.15, we changed the response format of the import endpoint to return JSONL for performance reasons, instead of JSON. I’ll update the JS client library today to handle this shortly for createMany, in the meantime if you split by \n you should get an array of results.

**Comment by fabien** [comment]
@jasonbosco thanks - so I guess the JS client library will still return proper arrays and objects, once updated to handle JSONL?

**Comment by jasonbosco** [comment]
That’s correct. We also added separate import & export methods that will return JSONL should you need the raw response, but createMany will return proper arrays and objects.

**Comment by jasonbosco** [comment]
@fabien I've updated the JS client in this pre-release version: https://www.npmjs.com/package/typesense/v/0.7.0-0

**Comment by fabien** [comment]
@jasonbosco thanks, that was fast!

**Comment by kishorenc** [comment]
Fixed in the v0.15.0 release.
